File: app/Http/Dto/v1/DataRoot.php
<?php

namespace Irail\Http\Dto\v1;

/* Copyright (C) 2011 by iRail vzw/asbl */

use AllowDynamicProperties;
use Exception;

/**
 * This is the root of every document. It will specify a version and timestamp. It also has the printer class to print the entire document.
 */
#[AllowDynamicProperties]
class DataRoot
{
    private $rootName;

    public $version;
    public $timestamp;

    /**
     * constructor of this class.
     *
     * @param        $rootname
     * @throws Exception
     * @internal param format $string the format of the document: json, json or XML
     */
    public function __construct($rootName)
    {
        $this->rootName = $rootName;
        $this->version = '1.3';
        $this->timestamp = date('U');
    }

    public function getRootName():string{
        return $this->rootName;
    }
}

File: src/main/java/be/irail/api/legacy/DataRoot.java
package be.irail.api.legacy;

/**
 * This is the root of every document. It will specify a version and timestamp.
 * It also has the printer class to print the entire document.
 */
public class DataRoot {
    private final String _rootName;
    public String version;
    public String timestamp;
    
    // Dynamic fields for different response types
    public Object station;
    public Object departure;
    public Object arrival;
    public Object connection;
    public Object vehicle;
    public Object stop;
    public Object disturbance;
    public Object composition;

    public DataRoot(String rootName) {
        this._rootName = rootName;
        this.version = "1.4";
        this.timestamp = String.valueOf(System.currentTimeMillis() / 1000);
    }

    public String getRootName() {
        return _rootName;
    }
}
